common/functions/src/scalars/maths/random.rs

```diff
@@ -0,0 +1,130 @@
+
+#[derive(Clone)]
+pub struct RandomFunction {
+    display_name: String,
+}
+
+impl RandomFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(RandomFunction {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default())
+    }
+}
+
+impl Function for RandomFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn num_arguments(&self) -> usize {
+        0
+    }
+
+    fn variadic_arguments(&self) -> Option<(usize, usize)> {
+        Some((0, 1))
+    }
+
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if args.is_empty()
+            || matches!(
+                args[0],
+                DataType::UInt8
+                    | DataType::UInt16
+                    | DataType::UInt32
+                    | DataType::UInt64
+                    | DataType::Null
+            )
+        {
+            Ok(DataType::Float64)
+        } else {
+            Err(ErrorCode::IllegalDataType(format!(
+                "Expected numeric types, but got {}",
+                args[0]
+            )))
+        }
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
+        let r_column: DataColumn = match columns.len() {
+            1 => {
+                let d_column: &DataColumn =
+                    &columns[0].column().cast_with_type(&DataType::UInt64)?;
+
+                match d_column {
+                    DataColumn::Constant(d, _) => {
+                        if d.is_null() {
+                            let mut rng = rand::thread_rng();
+                            DFFloat64Array::new_from_iter(
+                                (0..input_rows).into_iter().map(|_| rng.gen::<f64>()),
+                            )
+                        } else {
+                            let s: u64 = DFTryFrom::try_from(d.clone())?;
+                            let mut rng = rand::rngs::StdRng::seed_from_u64(s);
+                            DFFloat64Array::new_from_iter(
+                                (0..input_rows).into_iter().map(|_| rng.gen::<f64>()),
+                            )
+                        }
+                    }
+                    DataColumn::Array(d_series) => binary(
+                        &DFUInt64Array::new_from_iter(
+                            (0..input_rows).into_iter().map(|r| r as u64),
+                        ),
+                        d_series.u64()?,
+                        |_, s| {
+                            let mut rng = rand::rngs::StdRng::seed_from_u64(s);
+                            rng.gen::<f64>()
+                        },
+                    ),
```

Comment:
Using ArrayApply instead of binary for one iterator.

```suggestion
                   DataColumn::Array(seed_series) => seed_series.u64()?.apply_cast_numeric(|seed| {
                       let mut rng = StdRng::seed_from_u64(seed);
                       rng.gen::<f64>()
               }),
```

Reply:
make sense, applied in my commit